### PR TITLE
Number all type variants

### PR DESCRIPTION
**Pull request by FredTheDino**
Adds a macro so enums can be converted into numbers - very pain free.

### CODE REVIEW

File: sylt-common/src/ty.rs [hunk 1]
```diff
@@ -29,67 +33,43 @@ pub enum Type {
...
+            Type::Ty
+            | Type::Void
+            | Type::Unknown
+            | Type::Int
+            | Type::Float
+            | Type::Bool
+            | Type::ExternFunction(..)
+            | Type::String => {}
         }
         .hash(h);
```
Comment (sornas) [inline sylt-common/src/ty.rs:74]: Should this line still be here?

**Review by sornas — APPROVED** [review]
This looks good but I'm a bit confused about the PRs now. #255 looks like a duplicate  of this and I think the linked issue here (#169) should link to something else.

File: sylt-common/src/ty.rs (hunk 1 above)
Comment (FredTheDino) [reply sylt-common/src/ty.rs:74]: Yes

Comment (FredTheDino) [inline sylt-common/src/ty.rs:74]:
```suggestion
```

**Pull request merged**